### PR TITLE
Application Service Created

### PR DESCRIPTION
**Pull request by AshenWijesingha**

### CODE REVIEW

File: src/api/services/Application.service.ts [hunk 1]
```diff
@@ -1,17 +1,57 @@
...
+    .catch((error) => {
+        throw new Error(error.message);
+    });
+};
 
 /**
  * @todo create @function fetchApplications to fetch all the Applications in the system
  */
+export const fetchApplications = async () => {
+    return await ApplicationModel.find()
```
Comment (senuravihanjayadeva) [inline src/api/services/Application.service.ts:36]: soft deleted ones shouldnt be displayed. Please filter those soft deleted data. You can use match method for this

File: src/api/services/Application.service.ts [hunk 2]
```diff
@@ -1,17 +1,57 @@
...
+        throw new Error(error.message);
+    });
+};
 
 /**
  * @todo create @function archiveApplication to delete an Application
  * @param applicationId @type string
  */
+export const archiveApplication = async (applicationId: string) => {
+    return await ApplicationModel.findByIdAndDelete(applicationId)
```
Comment (senuravihanjayadeva) [inline src/api/services/Application.service.ts:50]: Please refer delete method in event service. Here we are updating only deletedAt attribute. Not a hard delete

**Review by senuravihanjayadeva — CHANGES_REQUESTED** [review]

File: src/api/services/Application.service.ts (hunk 1 above)
Comment (AshenWijesingha) [reply src/api/services/Application.service.ts:36]: ok ayye

File: src/api/services/Application.service.ts (hunk 2 above)
Comment (AshenWijesingha) [reply src/api/services/Application.service.ts]: Done ayye

File: src/api/services/Application.service.ts [hunk 3]
```diff
@@ -1,17 +1,63 @@
...
+    .catch((error) => {
+        throw new Error(error.message);
+    });
+};
 
 /**
- * @todo create @function fetchApplications to fetch all the Applications in the system
+ * @todo create @function fetchApplications to fetch all the Applications in the system.
  */
+export const fetchApplications = async (filter: FilterQuery<IApplication>) => {
```
Comment (senuravihanjayadeva) [inline src/api/services/Application.service.ts:35]: for fliter data which is deletedAt = null refer mongoose documentation
https://mongoosejs.com/docs/api.html#model_Model.find
ex :  MyModel.find({ name: 'john'} )

Comment (senuravihanjayadeva) [reply src/api/services/Application.service.ts:35]: remove filter in line 35 and 36

**Review by senuravihanjayadeva — CHANGES_REQUESTED** [review]

File: src/api/services/Application.service.ts [hunk 4]
```diff
@@ -1,17 +1,63 @@
+import { DocumentDefinition, FilterQuery } from "mongoose";
+import { IApplication} from "../interfaces";
+import ApplicationModel from "../models/Application.model";
+
 /**
  * @todo create @function addApplication to save an Application in the database
```
Comment (rusiruavb) [inline src/api/services/Application.service.ts:6]: Remove the `@todo` and add proper comment for the function.
```
/**
 * @param {IContact} applicationData
 * @returns {IContact} Application document
 */
```

File: src/api/services/Application.service.ts [hunk 5]
```diff
@@ -1,17 +1,63 @@
...
  */
+export const addApplication = async (applicationData: DocumentDefinition<IApplication>) => {
+  return await ApplicationModel.create(applicationData)
+    .then(async (application) => {
```
Comment (rusiruavb) [inline src/api/services/Application.service.ts:10]: No need to have `async` in here. Remove the `async` keyword

File: src/api/services/Application.service.ts [hunk 6]
```diff
@@ -1,17 +1,63 @@
...
+    });
+};
 
 /**
  * @todo create @function fetchApplicationById to fetch a Application  in the system
  * @param applicationId @type string
  */
+export const fetchApplicationById = async (applicationId: string) => { 
+    return await ApplicationModel.findById(applicationId)
+    .then(async (application) => {
```
Comment (rusiruavb) [inline src/api/services/Application.service.ts:24]: No need to have `async` in here. Remove the `async` keyword

File: src/api/services/Application.service.ts [hunk 7]
```diff
@@ -1,17 +1,63 @@
...
+    });
+};
 
 /**
- * @todo create @function fetchApplications to fetch all the Applications in the system
+ * @todo create @function fetchApplications to fetch all the Applications in the system.
  */
+export const fetchApplications = async (filter: FilterQuery<IApplication>) => {
+    return await ApplicationModel.find(filter)
+    .then(async (applications) => {
```
Comment (rusiruavb) [inline src/api/services/Application.service.ts:37]: No need to have `async` in here. Remove the `async` keyword

**Review by rusiruavb — CHANGES_REQUESTED** [review]

File: src/api/services/Application.service.ts (hunk 5 above)
Comment (AshenWijesingha) [reply src/api/services/Application.service.ts:10]: Done ayye

File: src/api/services/Application.service.ts (hunk 6 above)
Comment (AshenWijesingha) [reply src/api/services/Application.service.ts:24]: Done ayye

File: src/api/services/Application.service.ts (hunk 7 above)
Comment (AshenWijesingha) [reply src/api/services/Application.service.ts:37]: Done ayye

**Comment by AshenWijesingha** [comment]
requested changes applied

File: src/api/services/Application.service.ts [hunk 8]
```diff
@@ -1,17 +1,68 @@
...
+};
 
 /**
- * @todo create @function fetchApplications to fetch all the Applications in the system
+ * Application Service
+ * @param {FilterQuery<IApplication>} query
+ * @returns {Promise<IApplication>}
  */
+export const fetchApplications = async (filter: FilterQuery<IApplication>) => {
+    return await ApplicationModel.find({deleteAt: null})
```
Comment (senuravihanjayadeva) [inline src/api/services/Application.service.ts:41]: Are u sure is this working

Comment (senuravihanjayadeva) [reply src/api/services/Application.service.ts:41]: Refer fetchWebinars in Webinar service. Filter data according to that

**Review by senuravihanjayadeva — CHANGES_REQUESTED** [review]

File: src/api/services/Application.service.ts (hunk 8 above)
Comment (AshenWijesingha) [reply src/api/services/Application.service.ts:41]: OK ayye

File: src/api/services/Application.service.ts [hunk 9]
```diff
@@ -1,17 +1,68 @@
...
 
 /**
  * @todo create @function archiveApplication to delete an Application
  * @param applicationId @type string
  */
+export const archiveApplication = async (applicationId: string) => {
+    return await ApplicationModel.findById(applicationId)
+    .then(async (application) => {
+        if (application) {
+        application.deletedAt = new Date();
```
Comment (rusiruavb) [inline src/api/services/Application.service.ts:59]: Add proper indentation for the `if`` condition

File: src/api/services/Application.service.ts [hunk 10]
```diff
@@ -1,17 +1,68 @@
...
+
 /**
- * @todo create @function addApplication to save an Application in the database
+ * Application Service
+ * @param {IApplication} application
+ * @returns {Promise<IApplication>}
  */
+export const addApplication = async (applicationData: DocumentDefinition<IApplication>) => {
+  return await ApplicationModel.create(applicationData)
+    .then( (application) => {
```
Comment (rusiruavb) [inline src/api/services/Application.service.ts:12]: remove extra space between `then` and `(application)`
`then((application) => {})`

File: src/api/services/Application.service.ts [hunk 11]
```diff
@@ -1,17 +1,68 @@
...
 
 /**
- * @todo create @function fetchApplicationById to fetch a Application  in the system
+ * Application Service
  * @param applicationId @type string
+ * @returns {Promise<IApplication>}
  */
+export const fetchApplicationById = async (applicationId: string) => { 
+    return await ApplicationModel.findById(applicationId)
+    .then( (application) => {
```
Comment (rusiruavb) [inline src/api/services/Application.service.ts:27]: remove extra space between `then` and `(application)`
`then((application) => {})`

**Review by rusiruavb — CHANGES_REQUESTED** [review]

**Pull request closed without merging**